pkg/utils/utils.go

```diff
@@ -374,6 +375,17 @@ func configSriovDevice(iface *sriovnetworkv1.Interface, ifaceStatus *sriovnetwor
 	return nil
 }
 
+func generateRandomMacAddressVfs() ([]byte, error) {
+	buf := make([]byte, 6)
+	_, err := rand.Read(buf)
+	if err != nil {
+		return nil, err
+	}
+	// Set the local bit
+	buf[0] |= 2
```

Comment:
as far as i see. 

need to:
1. make sure mcast bit is 0 
2. local bit is 1

this only takes care of the latter

`buf[0] = (buf[0] & 0xfe) |  2`  will do the trick.

since you have 1 bit which is 0 and one bit with is 1  then the generated address cannot be all zeroes or all ones

Reply:
was looking at : 
<img src="https://en.wikipedia.org/wiki/MAC_address#/media/File:MAC-48_Address.svg">